### PR TITLE
Fix: Add Linux-compatible shell scripts for `run` and `run_continuous`

### PR DESCRIPTION
**Pull request by 0xf333**
### Background
This pull request addresses the need for Linux-compatible versions of the existing Windows batch files, `run.bat` and `run_continuous.bat`. The project's overall direction aims to enhance cross-platform compatibility and improve overall usability for users running Linux.

### Changes
- Created `run.sh` as the Linux equivalent of `run.bat`:
  - Checks and installs required Python packages if not already present.
  - Executes the `autogpt` module with any given command-line arguments.
  
- Created `run_continuous.sh` as the Linux equivalent of `run_continuous.bat`:
  - Adds the `--continuous` flag to the command-line arguments.
  - Calls the `run.sh` script with the modified arguments.

### Documentation
same as  PR #870 and #1678, this is just the Linux version. 

### Test Plan
1. Tested the scripts on a Linux system with Python 3 and pip3 installed.
2. Verified that the required Python packages were checked and installed correctly.
3. Checked that the `autogpt` module executed as expected with the provided command-line arguments.
4. Confirmed that the `--continuous` flag was added to the command-line arguments when running `run_continuous.sh`.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thoroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by 0xf333** [comment]
Hi @nponeccop,
Please review the new scripts when you have time and let me know if you have any feedback or suggestions for improvement.

File: run_continuous.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+argument="--continuous"
+./run.sh "$argument"
```
Comment (hdkiller) [inline run_continuous.sh]: If you pass “$@“ here then —help and other arguments will work for the run_continuous.sh script. Users might think that they passed arguments but those will be not sent to run.sh eventually to autogpt which might cause confusion.

Comment (nponeccop) [reply run_continuous.sh]: He just copies the original silliness in the Windows script lol, it's ok. The entire argument line is not necessary.

Comment (0xf333) [reply run_continuous.sh]: > If you pass “$@“ here then —help and other arguments will work for the run_continuous.sh script. Users might think that they passed arguments but those will be not sent to run.sh eventually to autogpt which might cause confusion.

@hdkiller & @nponeccop 

Thanks for the feedback, I appreciate it.
So basically I should change it to this: 
```
#!/bin/bash

./run.sh --continuous "$@"
```
This way the `--continuous` flag will be added to the command-line arguments, and any other arguments provided by the user will be passed through to the run.sh script, and eventually to the autogpt module.

Making the change now

Comment (nponeccop) [reply run_continuous.sh]: Should be --continuous $@ without double quotes, I guess?

**Comment by nponeccop** [comment]
@0xf333 There are conflicts now

**Comment by 0xf333** [comment]
> @0xf333 There are conflicts now

@nponeccop 

Conflicts fixed.
Thanks for the feedback, I appreciate it.

File: run_continuous.sh (hunk 1 above)
Comment (0xf333) [reply run_continuous.sh]: > Should be --continuous $@ without double quotes, I guess?

The quotes around "$@" are to preserve the original command-line arguments, including any arguments that contain spaces or other special characters. 

For instance, if you call `myscript.sh` with the following arguments:
```
./myscript.sh "arg1 with spaces" arg2 "arg3 with spaces"
```

Using "$@" will pass the arguments to `another_script.sh` exactly as they were received, preserving the spaces:

- arg1 with spaces
- arg2
- arg3 with spaces

If you were to use $@ without the quotes, the arguments would be split by spaces, and `another_script.sh` would receive them as separate arguments:

- arg1
- with
- spaces
- arg2
- arg3
- with
- spaces

Comment (nponeccop) [reply run_continuous.sh]: This is the intended behavior, isn't it? We just forward the arguments as is. If our original script received 15 arguments, the nested script receives 15, and not one big argument with spaces inside.

Comment (0xf333) [reply run_continuous.sh]: > This is the intended behavior, isn't it? We just forward the arguments as is. If our original script received 15 arguments, the nested script receives 15, and not one big argument with spaces inside.

Ok, adjusting it now

Comment (0xf333) [reply run_continuous.sh]: > This is the intended behavior, isn't it? We just forward the arguments as is. If our original script received 15 arguments, the nested script receives 15, and not one big argument with spaces inside.

done

**Pull request closed without merging**